code/game/gamemodes/miniantags/abduction/abduction.dm

```diff
@@ -0,0 +1,456 @@
+	agent.equip_to_slot_or_del(new /obj/item/clothing/under/color/grey(agent), slot_w_uniform) //they're greys gettit
+	agent.equip_to_slot_or_del(new /obj/item/weapon/storage/backpack(agent), slot_back)
+
+/datum/game_mode/abduction/proc/get_team_console(team)
+	var/obj/machinery/abductor/console/console
+	for(var/obj/machinery/abductor/console/c in abductor_equipment)
+		if(c.team == team)
+			console = c
+			break
+	return console
+
+/datum/game_mode/abduction/proc/equip_agent(mob/living/carbon/human/agent,team_number)
+	if(!team_number)
+		team_number = agent.mind.abductor.team
+
+	var/obj/machinery/abductor/console/console = get_team_console(team_number)
+	var/obj/item/clothing/suit/armor/abductor/vest/V = new /obj/item/clothing/suit/armor/abductor/vest(agent)
+	if(console!=null)
+		console.vest = V
+		V.flags |= NODROP
+	agent.equip_to_slot_or_del(V, slot_wear_suit)
+	agent.equip_to_slot_or_del(new /obj/item/weapon/abductor_baton(agent), slot_in_backpack)
+	agent.equip_to_slot_or_del(new /obj/item/weapon/gun/energy/decloner/alien(agent), slot_belt)
+	agent.equip_to_slot_or_del(new /obj/item/device/abductor/silencer(agent), slot_in_backpack)
+	agent.equip_to_slot_or_del(new /obj/item/clothing/head/helmet/abductor(agent), slot_head)
+	agent.update_icons()
+
+
+/datum/game_mode/abduction/proc/equip_scientist(var/mob/living/carbon/human/scientist,var/team_number)
+	if(!team_number)
+		team_number = scientist.mind.abductor.team
+
+	var/obj/machinery/abductor/console/console = get_team_console(team_number)
+	var/obj/item/device/abductor/gizmo/G = new /obj/item/device/abductor/gizmo(scientist)
+	if(console!=null)
+		console.gizmo = G
+		G.console = console
+	scientist.equip_to_slot_or_del(G, slot_in_backpack)
+
+	var/obj/item/weapon/implant/abductor/beamplant = new /obj/item/weapon/implant/abductor(scientist)
+	beamplant.implant(scientist)
+	scientist.update_icons()
+
+
+/datum/game_mode/abduction/check_finished()
+	if(!finished)
+		for(var/team_number=1,team_number<=abductor_teams,team_number++)
+			var/obj/machinery/abductor/console/con = get_team_console(team_number)
+			var/datum/objective/objective = team_objectives[team_number]
+			if (con.experiment.points >= objective.target_amount)
+				shuttle_master.emergency.request(null, 0.5)
+				finished = 1
+				return ..()
+	return ..()
+
+/datum/game_mode/abduction/declare_completion()
+	for(var/team_number=1,team_number<=abductor_teams,team_number++)
+		var/obj/machinery/abductor/console/console = get_team_console(team_number)
+		var/datum/objective/objective = team_objectives[team_number]
+		var/team_name = team_names[team_number]
+		if(console.experiment.points >= objective.target_amount)
+			to_chat(world, "<span class='greenannounce'>[team_name] team fullfilled its mission!</span>")
+		else
+			to_chat(world, "<span class='boldannounce'>[team_name] team failed its mission.</span>")
+	..()
+	return 1
+
+/datum/game_mode/proc/auto_declare_completion_abduction()
```

Comment:
can i just note how bad these auto_declare_completion procs are
like why does gamemode code do this
why not use an object oriented standard like auto_declare_completion() defined on each path
aaaaaaaaaahhhhhhhhhhhh